observer/probers/tls/tls_conf.go

```diff
@@ -42,15 +45,28 @@ func (c TLSConf) UnmarshalSettings(settings []byte) (probers.Configurer, error)
 }
 
 func (c TLSConf) validateHostname() error {
-	url, err := url.Parse(c.Hostname)
+	hostname := c.Hostname
+
+	if strings.Contains(c.Hostname, ":") {
+		host, port, err := net.SplitHostPort(c.Hostname)
+		if err != nil {
+			return fmt.Errorf("invalid 'hostname', got %q, expected a valid hostport: %s", c.Hostname, err)
+		}
+
+		_, err = strconv.Atoi(port)
+		if err != nil {
+			return fmt.Errorf("invalid 'hostname', got %q, expected a valid hostport: %s", c.Hostname, err)
+		}
+		hostname = host
+	}
+
+	url, err := url.Parse(hostname)
 	if err != nil {
-		return fmt.Errorf(
-			"invalid 'hostname', got %q, expected a valid hostname: %s", c.Hostname, err)
+		return fmt.Errorf("invalid 'hostname', got %q, expected a valid hostname: %s", c.Hostname, err)
```

Comment:
Same here

Reply:
No, I specifically want to use c.Hostname here, so that the error message reflects the actual input.